src/components/table/column-picker/column-picker.component.ts

```diff
@@ -203,14 +372,54 @@ export class ColumnPickerComponent {
             removeAllColumns: this.removeAllColumns.bind(this)
         };
     }
+
+    /** Change the expanded state of a node */
+    _setNodeExpanded(node: ColumnPickerTreeNode, isExpanded: boolean): void {
+        node.isExpanded = isExpanded;
+
+        // the first change detection cycle will hide the elements but we need to trigger
+        // a second change detection cycle on the next tick to ensure the ContentChildren
+        // QueryList gets updated in the uxTabbableList directive
+        requestAnimationFrame(() => this._changeDetectorRef.detectChanges());
+    }
 }
 
 /** Define a context for the column actions template */
 export interface ColumnPickerActionsContext {
-    addSelection: ReadonlyArray<string>;
-    removeSelection: ReadonlyArray<string>;
-    addColumns(columns?: ReadonlyArray<string>): void;
-    removeColumns(columns?: ReadonlyArray<string>): void;
+    addSelection: ReadonlyArray<string | ColumnPickerGroupItem>;
+    removeSelection: ReadonlyArray<string | ColumnPickerGroupItem>;
+    addColumns(columns?: ReadonlyArray<string | ColumnPickerGroupItem>): void;
+    removeColumns(columns?: ReadonlyArray<string | ColumnPickerGroupItem>): void;
     addAllColumns(): void;
     removeAllColumns(): void;
-}
\ No newline at end of file
+}
+
+/** An interface representing a grouped item */
+export interface ColumnPickerGroupItem {
+    // The name of the group that this column belongs to.
+    group?: string;
+    // The name of the column.
+    name: string;
+}
+
+/** An interface representing settings of groups defined in ColumnPickerGroupItem objects */
+export interface ColumnPickerGroup {
+    // The name of the group this setting object is related to.
+    name: string;
+    // Defines if this group will be expanded on load. This is an optional property.
+    expanded?: boolean;
+}
+
+/** Class representing a tree node item. Normalises data for both groups and columns into one format */
+export class ColumnPickerTreeNode {
```

Comment:
I think we also decided to change this to an interface to avoid any unnecessary increase to our bundle size